### PR TITLE
appActivation: Prevent showing windows before speedwagon

### PR DESCRIPTION
**Pull request by joaquimrocha**
When the overview is hidden, the actors corresponding to visible
(non-minimized) windows are shown. As a result, the speedwagon's
animation does show up as smooth as it should.

These changes hide the mentioned actors while the speedwagon is shown
and show them again just before the speedwagon disappears. This way the
speedwagon's animation is smooth and there is no impact on the behavior
after it is gone.

https://phabricator.endlessm.com/T5419

### CODE REVIEW

File: js/ui/appActivation.js [hunk 1]
```diff
@@ -139,8 +145,10 @@ const AppActivationContext = new Lang.Class({
     _maybeClearSplash: function() {
         // Clear the splash only when we've waited at least 700ms,
         // and when the app has transitioned to the running state...
-        if (this._appStateId == 0 && this._timeoutId == 0)
+        if (this._appStateId == 0 && this._timeoutId == 0) {
+            this._resetWindowsVisibility();
```
Comment (cosimoc) [inline js/ui/appActivation.js]: Should probably be done inside `_clearSplash()` as that's called from elsewhere too?

File: js/ui/appActivation.js [hunk 2]
```diff
@@ -56,6 +58,10 @@ const AppActivationContext = new Lang.Class({
         }
 
         if (showSplash) {
+            if (Main.overview.visible) {
+                this._hideWindows();
```
Comment (cosimoc) [inline js/ui/appActivation.js]: Should probably be done inside `this.showSplash()`, and only if the logic there actually decides to show the splash.
But not when `showSplash()` is called from `DesktopClient`, as that is only called when one application launches another (e.g. a file manager opening a file)

File: js/ui/appActivation.js [hunk 3]
```diff
@@ -150,6 +162,27 @@ const AppActivationContext = new Lang.Class({
...
+    _resetWindowsVisibility: function() {
+        for (let actor of this._hiddenWindows) {
+            actor.visible = true;
+        }
+
+        this._hiddenWindows = [];
+    },
+
+    _hideWindows: function() {
+        let windows = global.window_group.get_children();
```
Comment (cosimoc) [inline js/ui/appActivation.js]: You should use `shell_global_get_window_actors()`

File: js/ui/appActivation.js (hunk 1 above)
Comment (cosimoc) [inline js/ui/appActivation.js]: I don't think you need this anymore if you do it from `_clearSplash()`

Comment (joaquimrocha) [inline js/ui/appActivation.js]: I'd swear I moved it... Maybe it just copied it but I'm still surprised it got through. (I always use add -p for reviewing the changes I add)

**Pull request merged**